plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/navigator/DBNUtils.java

```diff
@@ -274,7 +276,14 @@ private static class NodeNameComparator implements Comparator<DBNNode> {
 
         @Override
         public int compare(DBNNode node1, DBNNode node2) {
-            return node1.getNodeDisplayName().compareToIgnoreCase(node2.getNodeDisplayName());
+            boolean isIgnoreCase = DBWorkbench.getPlatform().getPreferenceStore().getBoolean(ModelPreferences.NAVIGATOR_SORT_IGNORE_CASE);
+            String node1Name = node1.getNodeDisplayName();
+            String node2Name = node2.getNodeDisplayName();
+            if (isIgnoreCase) {
+                node1Name = node1Name.toUpperCase();
```

Comment:
It is much better to use compareToIgnoreCase than converting 2 strings to upper case.